src/HostManager.cpp

```diff
@@ -91,7 +91,7 @@ bool HostManager::addHost( QString hostname, QString port, QString login, QStrin
     }
 
     int portnumber = 23;
-    if( ! port.isEmpty() && port.toInt() > 0 && port.toInt() << 65536 )
+    if( ! port.isEmpty() && port.toInt() > 0 && port.toInt() < 65536 )
```

Comment:
So the old code left shifted the integer port number a very large number of places and then checked that it was non-zero - that sounds as though it would be unlikely for the number to ever read as `true` if I read say [this](http://stackoverflow.com/q/141525/4805858) right.